concurrency/buffer.go

```diff
@@ -5,18 +5,21 @@ import (
 	"sync"
 )
 
+// SyncBuffer is a struct that contains a buffer and a mutex lock to allow for synchronous access to the contained buffer.
 type SyncBuffer struct {
 	mu  sync.Mutex
 	buf bytes.Buffer
 }
 
+// Write will write bytes to SyncBuffer within a mutex lock.
```

Comment:
This comment duplicates code. I'd suggest to only keep comment on `SyncBuffer` itself, and remove the comments on methods.